assets/src/components/ladder.tsx

```diff
@@ -240,41 +240,35 @@ const orientationMatchingVehicle = (
   vehicleDirection === VehicleDirection.Down ? Orientation.Down : Orientation.Up
 
 const ScheduledLine = ({
-  ladderDirection,
-  ladderVehicle,
-  timepointStatusY,
+  ladderVehicle: {
+    status,
+    isUnassignedBySwiftly,
+    x,
+    y,
+    scheduledY,
+    scheduledVehicleDirection,
+  },
 }: {
-  ladderDirection: LadderDirection
   ladderVehicle: LadderVehicle
-  timepointStatusY: TimepointStatusYFunc
 }) => {
-  const { vehicle } = ladderVehicle
-  const { scheduleAdherenceStatus, scheduledLocation } = vehicle
-  if (scheduledLocation !== null) {
-    const scheduledVehicleDirection: VehicleDirection = directionOnLadder(
-      scheduledLocation.directionId,
-      ladderDirection
-    )
-    const scheduledY = timepointStatusY(
-      scheduledLocation.timepointStatus,
-      scheduledVehicleDirection
-    )
-    const roadLineX =
-      scheduledVehicleDirection === VehicleDirection.Up
-        ? CENTER_TO_LINE
-        : -CENTER_TO_LINE
-    return (
-      <line
-        className={`m-ladder__scheduled-line ${scheduleAdherenceStatus}`}
-        x1={ladderVehicle.x}
-        y1={ladderVehicle.y}
-        x2={roadLineX}
-        y2={scheduledY}
-      />
-    )
-  } else {
+  if (!scheduledY || isUnassignedBySwiftly) {
```

Comment:
It's needed because scheduledY is always set as long as we have a schedule (which is what the `!scheduledY` case is checking for). So this condition is combining two cases, if there is no schedule (in which case we can't draw a line), or if the trip is unassigned by Swiftly (in which case we are choosing not to draw a line).

Reply:
For off course vehicles, `y` is already the scheduled location. Can we not set `scheduledY` for off course vehicles?

Reply:
We could, but again I think that is conflating things in a bad way. "`sheduledY` is null not because we don't have it but because we removed it because that signifies a case where we don't want to draw a line" does not feel like a good way of representing this data to me.

Reply:
The viewmodel isn't supposed to represent the data, it's supposed to represent what we draw on the screen. (Renaming `y` -> `triangleY` and `scheduledY` -> `scheduledLineY` might make that more clear.)

Reply:
Yeah I hear what you are saying but I think that that is what it is doing and I like the current names.